doc/influxql-2.md

```diff
@@ -0,0 +1,910 @@
+
+| Series               |10 `host:a,dc:west`|10 `host:b,dc:east`|10 `host:c,dc:east`|10 `host:d,dc:west`|20 `host:a,dc:west`|20 `host:b,dc:east`|20 `host:c,dc:east`|20 `host:d,dc:west`|
+|----------------------|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+|`{"metric":"network"}`|  1  |  2  |  3  |  4  |  1  |  2  |  3  |  4  |
+
+Note that in results that get returned to the client, only the series information would be retained, unless we pass in an argument to keep certain keys as tagsets on the individual points.
+
+We can merge based on multiple tag keys. For example if we called `foo.merge(keys:["metric", "dc"]) we would get a matrix that looks like this:
+
+| Series                           |  10 `host:a`| 10 `host:b`| 10 `host:c`| 10 `host:d`| 20 `host:a`| 20 `host:b`| 20 `host:c`| 20 `host:d`|
+|----------------------------------|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+|`{"metric":"network","dc":"west"}`|  1  |     |     |  4  |  1  |     |     |  4  |
+|`{"metric":"network","dc":"east"}`|     |  2  |  3  |     |     |  2  |  3  |     |
+
+#### Keys
+`keys` will return an array of tag keys that occur in the matrix. The output of keys function can be used as the predecessor argument to `empty` or `not empty` in a `where` expression. For pagination through keys, `limit` can be chained off it.
+
+```javascript
+// Accepts: matrix
+// Arguments: none
+// Returns: []string
+select(database:"foo").keys()
+```
+
+#### Values
+`values` returns a string array of values in sorted order for a given key and the given criteria. For pagination through values, `limit` can be chained off of it.
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   key - required argument, a string of what tag key to return values for
+// Returns: []string
+select(database:"foo").values(key:"metric")
+```
+
+#### Cardinality
+`cardinality` returns a map of the count of unique tag values for each key in the matrix.
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   keys - optional string array. only return the cardinality for the passed keys
+// Returns: matrix
+select(database:"foo", where:{'region'="west"})
+  .cardinality()
+```
+
+#### Limit
+`limit` accepts either a matrix or an array and will limit the number of results. In the case of a matrix, it limits the number of series (or rows) that are returned. In the case of an array it is used to take a slice. It's a good practice to always apply a limit of some kind.
+
+```javascript
+// Accepts: matrix, array
+// Arguments:
+//   n - the number of results to limit to, required argument.
+//   offset - the offset (inclusive) to start with. defaults to 0.
+// Returns: a matrix if passed a matrix, an array if passed an array
+//
+select(database:"foo").limit(n:10,offset:20) // get the third page of results
+```
+
+#### Time Shift
+`timeShift` will shift all returned timestamps in the matrix by the given duration.
```

Comment:
Should this be renamed to simply `shift` since we are in a time domain? Or do we want the explicit name?